### PR TITLE
Apply solid principles

### PR DESCRIPTION
**Pull request by aimalamiri**
## Apply SOLID principles

Move the parent_permission from the `Person` class to the `Student` class because it was not used in the `Teacher` class.

Split the `App` into 3 separate new classes `ManagePeople`, `ManageBooks`, and `ManageRentals`.

### CODE REVIEW

File: classes/manage_books.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def add_book
+    print 'Enter the name of the book: '
+    name = gets.chomp.to_s
+    print 'Enter the author of the book: '
+    author = gets.chomp.to_s
+    book = Book.new(name, author)
+    @books << book
+    puts 'Book created successfully'
+  end
```
Comment (mikethreels) [inline classes/manage_books.rb:22]: - [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: classes/manage_people.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    name = gets.chomp.to_s
+    print 'Enter the age of the student: '
+    age = gets.chomp.to_i
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.to_s.downcase == 'y'
+    student = Student.new(age, name, parent_permission: parent_permission)
+    @persons << student
+
+    puts "Student with id #{student.id} created"
+  end
```
Comment (mikethreels) [inline classes/manage_people.rb:34]: - [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: classes/manage_people.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    name = gets.chomp.to_s
+    print 'Enter the age of the teacher: '
+    age = gets.chomp.to_i
+    print "And teacher's specialization: "
+    specilization = gets.chomp.to_s
+    teacher = Teacher.new(specilization, age, name)
+    @persons << teacher
+
+    puts "Student with id #{teacher.id} created"
+  end
```
Comment (mikethreels) [inline classes/manage_people.rb:47]: - [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: classes/manage_rentals.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    person = gets.chomp.to_i
+    puts 'Invalide input!' if !(person.is_a? Integer) && person >= @people.length
+
+    print 'Please enter the date: '
+    date = gets.chomp.to_s
+
+    Rental.new(date, @books[book], @persons[person])
+
+    puts 'Rental created successfully'
+  end
```
Comment (mikethreels) [inline classes/manage_rentals.rb:46]: - [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/manage_books.rb (hunk 1 above)
Comment (aimalamiri) [reply classes/manage_books.rb:22]: Hi @mikethreels! First of all, thank you so much for the nice feedback, but I have a question in here. What can be the possible solution to split the input and output from the business logic here? you may suggest another method for printing the output in the terminal, right? 

So if I create another method just for outputting the text, I'll still pass the output text to the method as a parameter which will become the same concept. And the same concept is true about getting the input.
According to a few years of experience that I have in software development, it's not possible to split the output and input from the business logic in this method, and creating another class or method is just a mess.

**Review by Lameck1 — COMMENTED** [review]
Hello @aimalamiri 👋,

_This is an additional review as a result of the second opinion request._

Apologies for any inconvenience caused. I would to confirm that your implementation of the project requirements meets expectations.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Review by arslanbisharat — APPROVED** [review]
Hi Team,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**